cloudep_prep.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash -ex
+
+# Extract JSON args into shell variables
+JQ=$(command -v jq || true)
+[[ -z "${JQ}" ]] && echo "ERROR: Missing command: 'jq'" >&2 && exit 1
+
+eval "$(${JQ} -r '@sh "ENDPOINT=\(.endpoint) PROJECT=\(.project)"')"
+
+function log() {
+    level=$1
+    msg=$2
+    echo "${LEVEL}: $msg" >&2
+}
+
+TMP_DIR=$(mktemp -d)
+function cleanup() {
+  rm -rf "${TMP_DIR}"
+}
+trap cleanup EXIT
+
+if [[ ! -z ${GOOGLE_CREDENTIALS+x} && ! -z ${GOOGLE_PROJECT+x} ]]; then
```

Comment:
Does this script work if I configured service account credentials in my provider block?

Reply:
Probably not, do you have a pattern that works with both?

Reply:
Sadly not (since the provider doesn't expose the key, and gcloud doesn't accept a token). For now the best I'd suggest is to make sure this doesn't fail entirely since it seems this script itself isn't always necessary.

Reply:
Ya it is conditional per the variable, it should ignore this block when you are using a service account for gcloud auth. The script mostly assumes you have gcloud properly configured to make the calls. This block is in place so that you can pass credentials when running in an automated framework like my test environment.